mono/metadata/sgen-mono.c

```diff
@@ -56,6 +56,13 @@ gboolean sgen_mono_xdomain_checks = FALSE;
 /* Functions supplied by the runtime to be called by the GC */
 static MonoGCCallbacks gc_callbacks;
 
+/* The total number of bytes allocated so far in program exection by all attached threads.
+ * This is not constantly syncrhonized, but only updated on each GC. */
+static guint64 bytes_allocated_attached = 0;
+
+/* Total bytes allocated so far in program exevution by detached threads */ 
+static guint64 bytes_allocated_detached = 0;
```

Comment:
I don't think these variables logically belong here. GC specific stuff should stay in sgen folder, and is for the best to keep sgen-mono as small as possible, since it serves as a glue.  Let's move them somewhere else, in sgen-alloc for example, expose them with extern in sgen-gc.h for the whole gc and have mono_gc_get_total_allocated_bytes  call sgen_gc_get_total_allocated_bytes which does the work. I think this would look slightly better.

Reply:
I've put the variables in sgen-alloc, but added accessor functions instead of exposing them with extern, if that is acceptable. I prefer to control access to them in this way.